checks/kube/aws-irsa.sh

```diff
@@ -0,0 +1,1431 @@
+PG_SERVICE_ACCOUNTS=""
+OS_SERVICE_ACCOUNTS=""
+get_service_account_name() {
+    component_list=$1
+    category=$2
+    service_accounts_map="{}"
+
+    # Iterate over each component
+    IFS=',' read -r -a components <<< "$component_list"
+    for component in "${components[@]}"; do
+
+        if [[ $component =~ $EXCLUDE_PATTERN ]]; then
+            echo "[INFO] Skipping excluded component: $component"
+            continue
+        fi
+
+        # Check if component is enabled in HELM_CHART_VALUES
+        enabled_value=$(echo "$HELM_CHART_VALUES" | jq -r --arg comp "$component" '.[$comp].enabled')
+        if [ "$enabled_value" == "null" ]; then
+            enabled_value=$(echo "$HELM_CHART_DEFAULT_VALUES" | jq -r --arg comp "$component" '.[$comp].enabled')
+        fi
+
+        if [[ "$enabled_value" == "false" ]]; then
+            echo "[INFO] Component $component is disabled, skipping verification."
+            continue
+        fi
+
+        # Retrieve serviceAccount name from HELM_CHART_VALUES
+        service_account_name=$(echo "$HELM_CHART_VALUES" | jq -r --arg comp "$component" '.[$comp].serviceAccount.name // empty')
+
+        # Use default naming if no serviceAccount name is found in values
+        if [[ -z "$service_account_name" ]]; then
+            service_account_name="${HELM_RELEASE_NAME}-${component}"
+            echo "[INFO] Component $component has no custom service account name ($component.serviceAccount.name), falling back on default: $service_account_name."
+        fi
+
+        # Add to JSON mapping
+        service_accounts_map=$(echo "$service_accounts_map" | jq --arg comp "$component" --arg sa "$service_account_name" '. + {($comp): {"serviceAccountName": $sa}}')
+    done
+
+    # Assign the result to the appropriate category variable
+    if [[ "$category" == "pg" ]]; then
+        PG_SERVICE_ACCOUNTS="$service_accounts_map"
+    elif [[ "$category" == "os" ]]; then
+        OS_SERVICE_ACCOUNTS="$service_accounts_map"
+    fi
+}
+
+# Retrieve and map service accounts for each component category
+if [[ -n "$HELM_CHART_VALUES" ]]; then
+    echo "[INFO] Creating service account mappings for components."
+
+    get_service_account_name "$COMPONENTS_TO_CHECK_IRSA_PG" "pg"
+    get_service_account_name "$COMPONENTS_TO_CHECK_IRSA_OS" "os"
+
+    echo "[INFO] PostgreSQL Components Service Account Mapping:"
+    if ! echo "$PG_SERVICE_ACCOUNTS" | jq .; then
+        echo "[ERROR] Failed to parse PostgreSQL service account mapping JSON. Please check the Helm chart values." 1>&2
+        exit 1
+    fi
+
+    echo "[INFO] OpenSearch Components Service Account Mapping:"
+    if ! echo "$OS_SERVICE_ACCOUNTS" | jq .; then
+        echo "[ERROR] Failed to parse OpenSearch service account mapping JSON. Please check the Helm chart values." 1>&2
+        exit 1
+    fi
+
+else
+    echo "[FAIL] Cannot retrieve Helm chart values; unable to check service accounts." 1>&2
+    exit 1
+fi
+
+check_connectivity_with_nmap() {
```

Comment:
I've seen this function being used by essentially every component.
Does it make sense to do so?
if it works for identity, why wouldn't it work for webModeler. It's nothing component specific as it just does an nmap to the same endpoint and port as the other calls as well.

It's not like an ephemeral debug container on an existing pod but an extra spawned pod that doesn't differ between components.

Maybe I'm overlooking something?

Reply:
makes sense, thanks!